We have this error in the CI:

```
======================================================================
FAIL: test_push_apply_delta_file_with_error (qfieldcloud.core.tests.test_delta.QfcTestCase.test_push_apply_delta_file_with_error)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/src/app/qfieldcloud/core/tests/test_delta.py", line 301, in test_push_apply_delta_file_with_error
    self.upload_and_check_deltas(
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        project=project,
        ^^^^^^^^^^^^^^^^
    ...<8 lines>...
        ],
        ^^
    )
    ^
  File "/usr/src/app/qfieldcloud/core/tests/test_delta.py", line 1285, in upload_and_check_deltas
    self.check_deltas_by_file_id(
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        project,
        ^^^^^^^^
    ...<5 lines>...
        immediate_values,
        ^^^^^^^^^^^^^^^^^
    )
    ^
  File "/usr/src/app/qfieldcloud/core/tests/test_delta.py", line 1361, in check_deltas_by_file_id
    self.fail(f"Got failing status {payload[idx]['status']}", job=job)
    ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/app/qfieldcloud/core/tests/test_delta.py", line 140, in fail
    super().fail(msg)
    ~~~~~~~~~~~~^^^^^
AssertionError: Got failing status STATUS_ERROR

JobType: delta_apply
Output:
================
::<<<::8ccfdfee-ded5-4fd7-83a2-398438ab5929 Start QGIS Application
10:59:55.440 root      INFO     Available user defined environment variables: []
10:59:55.441 root      INFO     Starting QGIS app version 34410 (565fe70bd9e)...
10:59:55.605 root      INFO     QGIS app started!
::>>>::8ccfdfee-ded5-4fd7-83a2-398438ab5929 2
::<<<::ae7ed42f-fedb-4c44-8e31-f06b66b56792 Download Project Directory
10:59:55.607 root      INFO     Preparing a temporary directory for project files…
10:59:55.607 root      INFO     Downloading project files to /tmp/tmpraqy563_/files…
10:59:55.654 root      INFO     Downloading project files…
10:59:55.743 /usr/local/lib/python3.12/dist-packages/qfieldcloud_sdk/sdk.py INFO     Downloading file "points.geojson"…
10:59:55.837 /usr/local/lib/python3.12/dist-packages/qfieldcloud_sdk/sdk.py INFO     Downloading file "polygons.geojson"…
10:59:55.899 /usr/local/lib/python3.12/dist-packages/qfieldcloud_sdk/sdk.py INFO     Downloading file "testdata.gpkg"…
10:59:55.958 /usr/local/lib/python3.12/dist-packages/qfieldcloud_sdk/sdk.py INFO     Downloading file "project.qgs"…
10:59:56.049 /usr/local/lib/python3.12/dist-packages/qfieldcloud_sdk/sdk.py INFO     Downloading file "nonspatial.csv"…
10:59:56.049 root      INFO     Downloading project files finished!
10:59:56.050 root      INFO     Getting project files list…
10:59:56.051 root      INFO     Local files list for project "3e94f974-a372-4a86-9fef-3722d374663a":
Name                Size  MD5 Checksum
----------------  ------  --------------------------------
nonspatial.csv        17  4edc8dae8dcbe698552b282c882babc1
points.geojson       629  6e48ef79159cce33ed327a5d21c170fe
polygons.geojson     604  e16aacaa84d6956f9304e1c6bc9bb7fc
project.qgs       183135  75baef4fcd41cadd3c376772922a4212
testdata.gpkg     176128  6bc473b7ae6c104fefa9c73effe5bcc7
::>>>::ae7ed42f-fedb-4c44-8e31-f06b66b56792 2
::<<<::ffc08c4b-9594-4d54-875f-f6d6877bb161 Apply Deltas
10:59:56.052 root      INFO     Loading project file "/tmp/tmpraqy563_/files/project.qgs"...
10:59:56.131 root      INFO     Loading delta file "/io/deltafile.json"...
10:59:56.135 qfc_worker.commands.apply_deltas WARNING  Error while applying a single delta: No layer with id "pluto"
::>>>::ffc08c4b-9594-4d54-875f-f6d6877bb161 1
10:59:56.152 root      INFO     Stopping QGIS app…
10:59:56.193 root      INFO     Deleted QGIS app!

================

Error stack:
================Traceback (most recent call last):
  File "/usr/src/app/qfc_worker/workflow.py", line 261, in run_workflow
    return_values = step.method(**arguments)
                    ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/app/qfc_worker/commands/apply_deltas.py", line 276, in apply_deltas
    all_applied = apply_deltas_without_transaction(
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/app/qfc_worker/commands/apply_deltas.py", line 613, in apply_deltas_without_transaction
    layer.commitChanges()
    ^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'commitChanges'
  'NoneType' object has no attribute 'commitChanges'
================

Feedback:
================
{
  "container_exit_code": 0,
  "error": "'NoneType' object has no attribute 'commitChanges'",
  "error_class": "AttributeError",
  "error_origin": "container",
  "error_stack": "Traceback (most recent call last):\n  File \"/usr/src/app/qfc_worker/workflow.py\", line 261, in run_workflow\n    return_values = step.method(**arguments)\n                    ^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/src/app/qfc_worker/commands/apply_deltas.py\", line 276, in apply_deltas\n    all_applied = apply_deltas_without_transaction(\n                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/src/app/qfc_worker/commands/apply_deltas.py\", line 613, in apply_deltas_without_transaction\n    layer.commitChanges()\n    ^^^^^^^^^^^^^^^^^^^\nAttributeError: 'NoneType' object has no attribute 'commitChanges'\n",
  "error_type": "UNKNOWN",
  "feedback_version": "2.0",
  "outputs": {
    "download_project_directory": {},
    "start_qgis_app": {}
  },
  "steps": [
    {
      "id": "start_qgis_app",
      "name": "Start QGIS Application",
      "returns": {
        "qgis_version": "3.44.10-Solothurn"
      },
      "stage": 2
    },
    {
      "id": "download_project_directory",
      "name": "Download Project Directory",
      "returns": {
        "tmp_project_dir": "<non-serializable: PosixPath /tmp/tmpraqy563_>"
      },
      "stage": 2
    },
    {
      "id": "apply_deltas",
      "name": "Apply Deltas",
      "returns": {},
      "stage": 1
    },
    {
      "id": "stop_qgis_app",
      "name": "Stop QGIS Application",
      "returns": {},
      "stage": 0
    },
    {
      "id": "upload_project_directory",
      "name": "Upload Project",
      "returns": {},
      "stage": 0
    }
  ],
  "workflow_id": "apply_changes",
  "workflow_name": "Apply Changes",
  "workflow_version": "3.0"
}
================

----------------------------------------------------------------------
```

--- a/docker-qgis/qfc_worker/commands/apply_deltas.py
+++ b/docker-qgis/qfc_worker/commands/apply_deltas.py
@@ -413,7 +413,6 @@ def apply_deltas_without_transaction(
     for idx, delta in enumerate(delta_file.deltas):
         delta_status = DeltaStatus.Applied
         layer_id: str = delta.get("sourceLayerId", "")
-        layer_ids_to_commit.append(layer_id)
         layer: QgsVectorLayer = project.mapLayer(layer_id)
         feature = QgsFeature()
 
@@ -429,6 +428,9 @@ def apply_deltas_without_transaction(
                     f'Cannot start editing layer "{layer_id}"',
                     provider_errors=layer.dataProvider().errors(),
                 )
+
+            if layer_id not in layer_ids_to_commit:
+                layer_ids_to_commit.append(layer_id)
 
             # check if a PostGIS layer's session_role override is requested.
             if layer.providerType() == "postgres" and QFC_PG_EFFECTIVE_USER:

--- a/docker-qgis/qfc_worker/commands/apply_deltas.py
+++ b/docker-qgis/qfc_worker/commands/apply_deltas.py
@@ -407,11 +407,13 @@ def apply_deltas_without_transaction(
     overwrite_conflicts: bool = False,
 ) -> bool:
     has_applied_all_deltas = True
+    layer_ids_to_commit = []
 
     # apply deltas on each individual layer
     for idx, delta in enumerate(delta_file.deltas):
         delta_status = DeltaStatus.Applied
         layer_id: str = delta.get("sourceLayerId", "")
+        layer_ids_to_commit.append(layer_id)
         layer: QgsVectorLayer = project.mapLayer(layer_id)
         feature = QgsFeature()
 
@@ -495,7 +497,7 @@ def apply_deltas_without_transaction(
                 # in QGIS the only way to get the real features that have been added after commit, if edit buffer is present, is to use this signal.
                 layer.committedFeaturesAdded.connect(committed_features_added_cb)
 
-            if not layer.commitChanges():
+            if not layer.commitChanges(False):
                 raise DeltaException(
                     "Failed to commit changes",
                     provider_errors=layer.dataProvider().errors(),
@@ -605,6 +607,10 @@ def apply_deltas_without_transaction(
             )
 
             raise err
+
+    for layer_id in layer_ids_to_commit:
+        layer = project.mapLayer(layer_id)
+        layer.commitChanges()
 
     return has_applied_all_deltas
 

--- a/docker-qgis/qfc_worker/commands/apply_deltas.py
+++ b/docker-qgis/qfc_worker/commands/apply_deltas.py
@@ -610,6 +610,7 @@ def apply_deltas_without_transaction(
 
             raise err
 
+    # Exits editing mode for modified layers, there is nothing to commit at this stage
     for layer_id in layer_ids_to_commit:
         layer = project.mapLayer(layer_id)
         layer.commitChanges()

